indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/CryptoPublicKey.kt

```diff
@@ -165,52 +165,34 @@ sealed class CryptoPublicKey : Asn1Encodable<Asn1Sequence>, Identifiable {
 
     }
 
-    /**
-     * RSA Public key
-     */
-    @Serializable
+    /** RSA Public key */
     @ConsistentCopyVisibility
     data class RSA
     @Throws(IllegalArgumentException::class)
-    private constructor(
-        /**
-         * RSA key size
-         */
-        val bits: Size,
-
-        /**
-         * modulus
-         */
-        @Serializable(with = ByteArrayBase64Serializer::class) val n: ByteArray,
+    constructor(
+        /** modulus */
+        val n: Asn1Integer.Positive,
 
-        /**
-         * public exponent
-         */
-        val e: Int,
+        /** public exponent */
+        val e: Asn1Integer.Positive,
     ) : CryptoPublicKey() {
 
-        init {
-            val computed = Size.of(n)
-            if (bits != computed) throw IllegalArgumentException("Provided number of bits (${bits.number}) does not match computed number of bits (${computed.number})")
-        }
-
-        @Throws(IllegalArgumentException::class)
-        private constructor(params: RsaParams) : this(
-            params.size,
-            params.n,
-            params.e
-        )
+        val bits = n.bitLength().let { Size.of(it) ?: throw IllegalArgumentException("Unsupported key size $it bits") }
 
         /**
          * @throws IllegalArgumentException in case of illegal input (odd key size, for example)
          */
         @Throws(IllegalArgumentException::class)
-        constructor(n: ByteArray, e: Int) : this(sanitizeRsaInputs(n, e))
+        constructor(n: ByteArray, e: ByteArray) : this(Asn1Integer.fromUnsignedByteArray(n), Asn1Integer.fromUnsignedByteArray(e))
```

Comment:
Since we're now using ASN.1 integer, we should document that the byte array contains only the magnitude

Reply:
i removed this constructor again (it was added in this PR) since it's not used anywhere, in favor of more obvious constructors involving BigInt